OS supported

Which OS does this library support? IIRC Windows doesn't allow replacing binaries at runtime because they get locked when a program uses them. 

My wild guess would be that `self_update` doesn't support Windows.

Example code has `ZipError: Specified file not found in archive` on Windows 10 x64.
```
...
Checking target-arch... x86_64-pc-windows-msvc
Checking current version... v0.5.1
Checking latest released version... v9.9.10
New release found! v0.5.1 --> v9.9.10
New release is *NOT* compatible

github release status:
  * Current exe: "C:\\Users\\artan\\CLionProjects\\self_update\\target\\debug\\examples\\github.exe"
  * New exe release: "self_update-v9.9.10-x86_64-pc-windows-msvc.zip"
  * New exe download url: "https://github.com/jaemk/self_update/releases/download/v9.9.10/self_update-v9.9.10-x86_64-pc-windows-msvc.zip"

The new release will be downloaded/extracted and the existing binary will be replaced.
Do you want to continue? [Y/n] Y
Downloading...
1.09 MB / 1.09 MB [=====================================] 100.00 % 481.01 KB/s   ... Done
Extracting archive... [ERROR] ZipError: Specified file not found in archive
error: process didn't exit successfully: `target\debug\examples\github.exe` (exit code: 1)

Process finished with exit code 1
```
Changing `.bin_name("github")` to `.bin_name("github.exe")` works well, successfully update executable.
```powershell
...
PS C:\Users\artan\CLionProjects\self_update\target\debug\examples> .\github.exe
Checking target-arch... x86_64-pc-windows-msvc
Checking current version... v9.9.10
Checking latest released version... v9.9.10
Update status: `9.9.10`!
```

--- a/src/backends/github.rs
+++ b/src/backends/github.rs
@@ -1,7 +1,7 @@
 /*!
 GitHub releases
 */
-use std::env;
+use std::env::{self, consts::EXE_SUFFIX};
 use std::path::{Path, PathBuf};
 
 use hyper_old_types::header::{LinkValue, RelationType};
@@ -276,9 +276,11 @@ impl UpdateBuilder {
     }
 
     /// Set the exe's name. Also sets `bin_path_in_archive` if it hasn't already been set.
+    ///
+    /// This method does append platform specific executable file suffixes to the name if needed.
     pub fn bin_name(&mut self, name: &str) -> &mut Self {
-        let raw_bin_name = format!("{}{}", name, EXECUTABLE_SUFFIX);
-        self.bin_name = Some(raw_bin_name.to_owned());
+        let raw_bin_name = format!("{}{}", name.trim_end_matches(EXE_SUFFIX), EXE_SUFFIX);
+        self.bin_name = Some(raw_bin_name.clone());
         if self.bin_path_in_archive.is_none() {
             self.bin_path_in_archive = Some(PathBuf::from(raw_bin_name));
         }

--- a/src/backends/github.rs
+++ b/src/backends/github.rs
@@ -277,9 +277,10 @@ impl UpdateBuilder {
 
     /// Set the exe's name. Also sets `bin_path_in_archive` if it hasn't already been set.
     pub fn bin_name(&mut self, name: &str) -> &mut Self {
-        self.bin_name = Some(name.to_owned());
+        let raw_bin_name = format!("{}{}", name, EXECUTABLE_SUFFIX);
+        self.bin_name = Some(raw_bin_name.to_owned());
         if self.bin_path_in_archive.is_none() {
-            self.bin_path_in_archive = Some(PathBuf::from(name));
+            self.bin_path_in_archive = Some(PathBuf::from(raw_bin_name));
         }
         self
     }

